### ISSUE TITLE
Report transfer values from iperf by pscheduler-tool-iperf3 (and iperf2)

### ISSUE THREAD

**Issue by tonin** [perfsonar/pscheduler#396]
With pS 3.5 and bwctl, psUI was making use of the **Transfer** values reported by iperf:

```
[ ID] Interval           Transfer     Bandwidth       Retr  Cwnd
[ 16]   0.00-1.00   sec  53.2 MBytes   445 Mbits/sec    0   60.8 KBytes       
[ 16]   1.00-2.00   sec  79.1 MBytes   666 Mbits/sec    0    198 KBytes       
[ 16]   2.00-3.00   sec  77.0 MBytes   646 Mbits/sec    0    273 KBytes       
[ 16]   3.00-4.00   sec  65.4 MBytes   547 Mbits/sec    5    219 KBytes       
[ 16]   4.00-5.01   sec  38.8 MBytes   324 Mbits/sec    0    219 KBytes       
[ 16]   5.01-6.01   sec  55.4 MBytes   465 Mbits/sec  123    177 KBytes       
[ 16]   6.01-7.00   sec  64.6 MBytes   545 Mbits/sec    0    189 KBytes       
[ 16]   7.00-8.01   sec  64.9 MBytes   540 Mbits/sec    0    195 KBytes       
[ 16]   8.01-9.01   sec  69.7 MBytes   584 Mbits/sec    0    199 KBytes       
[ 16]   9.01-10.02  sec  38.8 MBytes   323 Mbits/sec    0    199 KBytes       
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bandwidth       Retr
[ 16]   0.00-10.02  sec   607 MBytes   508 Mbits/sec  128             sender
[ 16]   0.00-10.02  sec   607 MBytes   508 Mbits/sec                  receiver
```

With pscheduler, that information is not available anymore:
```* Stream ID 4
Interval       Throughput     Retransmits    Current Window 
0.0 - 1.0      185.51 Mbps    0              2.07 MBytes    
1.0 - 2.0      178.33 Mbps    0              2.07 MBytes    
2.0 - 3.0      188.75 Mbps    0              2.07 MBytes    
3.0 - 4.0      188.74 Mbps    0              2.07 MBytes    
4.0 - 5.0      188.74 Mbps    0              2.07 MBytes    
5.0 - 6.0      188.74 Mbps    0              2.07 MBytes    
6.0 - 7.0      188.74 Mbps    0              2.07 MBytes    
7.0 - 8.0      178.26 Mbps    0              2.07 MBytes    
8.0 - 9.0      188.74 Mbps    0              2.07 MBytes    
9.0 - 10.0     188.75 Mbps    0              2.07 MBytes    
10.0 - 11.0    188.75 Mbps    0              2.07 MBytes    
11.0 - 12.0    188.74 Mbps    0              2.07 MBytes    
12.0 - 13.0    188.74 Mbps    0              2.07 MBytes    
13.0 - 14.0    188.74 Mbps    0              2.07 MBytes    
14.0 - 15.0    188.74 Mbps    0              2.07 MBytes    
15.0 - 16.0    178.26 Mbps    0              2.07 MBytes    
16.0 - 17.0    188.74 Mbps    0              2.07 MBytes    
17.0 - 18.0    188.75 Mbps    0              2.07 MBytes    
18.0 - 19.0    178.26 Mbps    0              2.07 MBytes    
19.0 - 20.0    188.74 Mbps    0              2.07 MBytes    

Summary
Interval       Throughput     Retransmits    
0.0 - 20.0     186.49 Mbps    0
```

Would it be possible to add that back or is there a particular reason these values are not reported anymore?

**Comment by daldoyle** [comment]
We could certainly add it back. I'd have to double check which tools actually report that - part of the goal here was to standardize the output of the various tools, so it may have been removed if say nuttcp didn't report it.

fwiw if you know the interval and the bandwidth the transfer is just basic math. (bandwidth / 8) * interval = transfer volume. There appears to be some rounding going on that makes those numbers not line up precisely but that basically gets you in the right ballpark.

Dan Doyle
GlobalNOC Software Developer
1-812-856-3892
daldoyle@globalnoc.iu.edu

**Comment by tonin** [comment]
Yes, it was to avoid getting inaccurate rounded values that I'd prefer to have those passed from iperf to the pscheduler client directly.  When 2 different tools are doing roundings you always end up wondering which one got it right.  I can tell that both iperf2 and iperf3 are reporting those transfer values, but I haven't tested with nuttcp.

**Comment by tonin** [comment]
Would it also be possible to have both the sender and receiver summary lines?

**Comment by daldoyle** [comment]
I have added the throughput volumes as reported by nuttcp, iperf3, and iperf2, will be available in 4.0.2. I still have to fix a few things in the bwctl tests since 4.0.2 was announced, had originally planned to just drop support for those and not worry about it.

Also still need to look into sender vs receiver summary lines.

**Comment by daldoyle** [comment]
This is all here minus the sender/receiver info:
https://github.com/perfsonar/pscheduler/pull/523

I'll make a separate issue on that, going to close this one out.
